app/models/shelter.rb

```diff
@@ -31,4 +32,13 @@ def alphabetical_pets
   def shelter_pets_filtered_by_age(age_filter)
     adoptable_pets.where("age >= ?", age_filter)
   end
+
+  # RAW SQL
+  def self.order_by_reverse_alphabetical
+    find_by_sql("SELECT * FROM shelters ORDER BY name DESC")
+  end
+
+  def self.shelters_with_pending_applications
+    Shelter.joins(pets: { application_pets: :application }).where(applications: { status: 'Pending' }).distinct.pluck(:name)
```

Comment:
Because you have the additional association to applications, you can also do `Shelter.joins(:applications)` directly and you should see the SQL that is used under the hood is doing the join through pets automatically  because you've created this shortcut with model associations